packages/rum-core/src/domain/resource/matchRequestTiming.ts

```diff
@@ -32,18 +36,24 @@ export function matchRequestTiming(request: RequestCompleteEvent) {
   }
 
   const candidates = sameNameEntries
-    .map((entry) => entry.toJSON() as RumPerformanceResourceTiming)
-    .filter(toValidEntry)
+    .filter((entry) => !matchedResourceTimingEntries.has(entry))
```

Comment:
<!-- Suggestions are specific requests to improve the subject under review. It is assumed that we all want to do what's best, so these comments are never dismissed as “mere suggestions”, but are taken seriously.  -->
**💬 suggestion:** ‏what about updating the function comment explaining the strategy with this additional behaviour?